src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,36 +1,138 @@
-import React from 'react';
+import React, {useState} from 'react';
 
-import { Button, ButtonProps } from '@patternfly/react-core';
+import {
+  ButtonProps,
+  Dropdown,
+  MenuToggle,
+  MenuToggleElement,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem, Flex, FlexItem, Spinner, MenuToggleAction,
+} from '@patternfly/react-core';
 
 import { selectSelectedBlueprintId } from '../../store/BlueprintSlice';
 import { useAppSelector } from '../../store/hooks';
-import { useComposeBlueprintMutation } from '../../store/imageBuilderApi';
+import {
+  ImageTypes,
+  useComposeBlueprintMutation,
+  useGetBlueprintQuery,
+} from '../../store/imageBuilderApi';
+import {EditSaveButton} from "../CreateImageWizardV2/steps/Review/Footer/EditDropdown";
+import {skipToken} from "@reduxjs/toolkit/query";
 
-type BuildImagesButtonPropTypes = {
-  variant?: ButtonProps['variant'];
-  children?: React.ReactNode;
-};
-
-export const BuildImagesButton = ({
-  variant,
-  children,
-}: BuildImagesButtonPropTypes) => {
+export const BuildImagesButton = () => {
   const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
+  const [selectedTargets, setSelectedTargets] =useState<ImageTypes[]>([]);
   const [buildBlueprint, { isLoading: imageBuildLoading }] =
     useComposeBlueprintMutation();
   const onBuildHandler = async () => {
-    selectedBlueprintId &&
-      (await buildBlueprint({ id: selectedBlueprintId, body: {} }));
+    debugger;
+    if (selectedBlueprintId) {
+      await buildBlueprint({
+        id: selectedBlueprintId,
+        body: {
+          image_types: selectedTargets
+        },
+      })};
+  };
+  const [selectedItems, setSelectedItems] = React.useState<number[]>([]);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | number | undefined
+  ) => {
+    setIsOpen(false);
+  };
+    const {
+      data: blueprintDetails,
+    } =  useGetBlueprintQuery(selectedBlueprintId ? { id: selectedBlueprintId } : skipToken)
+    const blueprintImageType = blueprintDetails?.image_requests.map(
+        (image) => image.image_type
+    );
+
+  const onSelect2 = (
```

Comment:
Let's find out more descriptive name - maybe `onSelectTarget`? Or whatever you think will be more fitting :)